File: liqp-core/src/main/java/liqp/nodes/OrNode.java
package liqp.nodes;

import liqp.node.LNode;
import lombok.Getter;

@Getter
public class OrNode extends ExpressionNode {

  public OrNode(LNode lhs, LNode rhs) {
    super(lhs, rhs);
  }

  @Override
  public Object render(LContext context) {
    return context.isTrue(lhs.render(context)) || context.isTrue(rhs.render(context));
  }
}

File: liqp-core/src/main/kotlin/liqp/nodes/OrNode.kt
package liqp.nodes

import liqp.context.LContext
import liqp.node.LNode

class OrNode(lhs: LNode, rhs: LNode) : ExpressionNode(lhs, rhs) {
  override fun render(context: LContext): Any {
    return context.isTrue(lhs.render(context)) || context.isTrue(rhs.render(context))
  }
}
